csa/connset.py

```diff
@@ -455,32 +455,31 @@ def __init__ (self, op1, op2):
     def iterator (self, low0, high0, low1, high1, state):
         iter1 = self.op1.iterator (low0, high0, low1, high1, state)
         iter2 = self.op2.iterator (low0, high0, low1, high1, state)
-        (i1, j1) = iter1.next ()
-        (i2, j2) = iter2.next ()
+        (i1, j1) = next(iter1)
+        (i2, j2) = next(iter2)
         while True:
             if (j1, i1) < (j2, i2):
                 yield (i1, j1)
-                (i1, j1) = iter1.next ()
+                (i1, j1) = next(iter1)
                 continue
             elif (i1, j1) == (i2, j2):
-                (i1, j1) = iter1.next ()
+                (i1, j1) = next(iter1)
             try:
-                (i2, j2) = iter2.next ()
+                (i2, j2) = next(iter2)
             except StopIteration:
                 while True:
                     yield (i1, j1)
-                    (i1, j1) = iter1.next ()
+                    (i1, j1) = next(iter1)
 
 
 def cmpPostOrder (c0, op1):
-    return cmp ((c0[1], c0[0]), (op1[1], op1[0]))
-
+    return  ((c0[1], c0[0]) > (op1[1], op1[0])) -  ((c0[1], c0[0]) < (op1[1], op1[0]))
 
 class ExplicitMask (FiniteMask):
     def __init__ (self, connections):
         FiniteMask.__init__ (self)
         self.connections = list (connections)
-        self.connections.sort (cmpPostOrder)
+        self.connections.sort (reverse=True)
```

Comment:
Maybe I'm missing something here, but this won't give us a post order sorting, right? Post order is _increasing_ but with the second element of a pair being most significant (as opposed to standard).